### PR TITLE
fix: appBar type to PreferredSizeWidget

### PR DESCRIPTION
**Pull request by Arenukvern**
Hi,

Thank you for the awesome library!

I would like to introduce small change - change type for appBar to PreferredSizeWidget.

Will glad for your review.

My best regards,
Anton

### CODE REVIEW

**Comment by d3xt3r2909** [comment]
https://github.com/d3xt3r2909/linkedin_login/issues/23

**Comment by d3xt3r2909** [comment]
Thanks for contribution, will take a look tomorrow (today).

File: example/pubspec.lock [hunk 1]
```diff
@@ -150,7 +150,7 @@ packages:
       name: material_color_utilities
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.1.2"
+    version: "0.1.3"
```
Comment (d3xt3r2909) [inline example/pubspec.lock:153]: Is this change needed as you didn't touch .yml please revert. This file should be updated in sync with upgrading dependencies

File: lib/src/server/linked_in_auth_code_widget.dart [hunk 2]
```diff
@@ -24,11 +24,11 @@ class LinkedInAuthCodeWidget extends StatefulWidget {
     this.useVirtualDisplay = false,
   });
 
-  final Function(AuthorizationSucceededAction)? onGetAuthCode;
-  final Function(AuthorizationFailedAction) onError;
+  final ValueChanged<AuthorizationSucceededAction>? onGetAuthCode;
```
Comment (d3xt3r2909) [inline lib/src/server/linked_in_auth_code_widget.dart:27]: Nice catch, but please define type on top of file to be more readable

File: pubspec.lock [hunk 3]
```diff
@@ -283,7 +283,7 @@ packages:
       name: material_color_utilities
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.1.2"
+    version: "0.1.3"
```
Comment (d3xt3r2909) [inline pubspec.lock:286]: Don't see need for this change

File: lib/src/server/linked_in_auth_code_widget.dart [hunk 4]
```diff
@@ -24,11 +24,11 @@ class LinkedInAuthCodeWidget extends StatefulWidget {
...
+  final ValueChanged<AuthorizationFailedAction> onError;
   final String? redirectUrl;
   final String? clientId;
-  final AppBar? appBar;
+  final PreferredSizeWidget? appBar;
```
Comment (d3xt3r2909) [inline lib/src/server/linked_in_auth_code_widget.dart:31]: Could you please add  tests which will test type of this parameter (did find widget with specific type). Thanks

**Review by d3xt3r2909 — CHANGES_REQUESTED** [review]

File: lib/src/server/linked_in_auth_code_widget.dart (hunk 2 above)
Comment (Arenukvern) [reply lib/src/server/linked_in_auth_code_widget.dart:27]: This is default Foundation typedef - please see more here https://api.flutter.dev/flutter/foundation/ValueChanged.html

File: example/pubspec.lock (hunk 1 above)
Comment (Arenukvern) [reply example/pubspec.lock]: done

File: pubspec.lock (hunk 3 above)
Comment (Arenukvern) [reply pubspec.lock]: done

File: lib/src/server/linked_in_auth_code_widget.dart (hunk 4 above)
Comment (Arenukvern) [reply lib/src/server/linked_in_auth_code_widget.dart:31]: Added, could you please review?

File: lib/src/server/linked_in_auth_code_widget.dart (hunk 2 above)
Comment (d3xt3r2909) [reply lib/src/server/linked_in_auth_code_widget.dart:27]: yes yes, but if we define it as our own type we will no any more need to change it in several places. 

For eg. you didn't change it inside `linked_in_auth_code_widget_test.dart` to be `ValueChanged` only because you were not aware of it, but if we make it something like 

`typedef OnGetAuthCode = void Function(AuthorizationSucceededAction);`

We will force our typing.

File: test/widget/src/server/linked_in_auth_code_widget_test.dart [hunk 5]
```diff
@@ -37,14 +37,14 @@ void main() {
       context: anyNamed('context'),
       creationParams: anyNamed('creationParams'),
       webViewPlatformCallbacksHandler:
-      anyNamed('webViewPlatformCallbacksHandler'),
+          anyNamed('webViewPlatformCallbacksHandler'),
```
Comment (d3xt3r2909) [inline test/widget/src/server/linked_in_auth_code_widget_test.dart:40]: please use auto-formated

File: test/widget/src/server/linked_in_auth_code_widget_test.dart [hunk 6]
```diff
@@ -115,4 +115,37 @@ void main() {
...
+    );
+
+    await tester.pumpWidget(testWidget);
+    await tester.pumpAndSettle();
+
+    expect(
+      find.byWidgetPredicate((widget) => widget is PreferredSizeWidget),
+      findsOneWidget,
+    );
+  });
```
Comment (d3xt3r2909) [inline test/widget/src/server/linked_in_auth_code_widget_test.dart:135]: add line please

File: test/widget/src/server/linked_in_auth_code_widget_test.dart [hunk 7]
```diff
@@ -115,4 +115,37 @@ void main() {
...
+      child: linkedInAuthCodeWidget(
+        appBar: AppBar(),
+      ),
+    );
+
+    await tester.pumpWidget(testWidget);
+    await tester.pumpAndSettle();
+
+    expect(
+      find.byWidgetPredicate((widget) => widget is PreferredSizeWidget),
```
Comment (d3xt3r2909) [inline test/widget/src/server/linked_in_auth_code_widget_test.dart:132]: please use finding by type as there is no callback that is not needed, but also its more readable

File: test/widget/src/server/linked_in_auth_code_widget_test.dart [hunk 8]
```diff
@@ -115,4 +115,37 @@ void main() {
 
     expect(find.text('AppBar title'), findsOneWidget);
   });
+
+  testWidgets('has PreferredSizeWidget type, if App Bar is provided', (
```
Comment (d3xt3r2909) [inline test/widget/src/server/linked_in_auth_code_widget_test.dart:119]: there is already test which is testing almost same thing `App bar is shown if it is provided`

Comment (d3xt3r2909) [reply test/widget/src/server/linked_in_auth_code_widget_test.dart:119]: what I meant by adding a test is to create a dummy custom AppBar widget which you are able to provide to testable widget and find by newly created custom type widget. (Sorry for misunderstanding I was typing from phone)

**Review by d3xt3r2909 — CHANGES_REQUESTED** [review]

File: lib/src/server/linked_in_auth_code_widget.dart (hunk 2 above)
Comment (d3xt3r2909) [reply lib/src/server/linked_in_auth_code_widget.dart:27]: Please ignore my comments regarding this. I have changed everything to the `ValueChanged` inside https://github.com/d3xt3r2909/linkedin_login/pull/72/files as it's required from `flutter_lint` itself. Anyhow maybe in the future, we could enforce this to be type-oriented but for now, let's go with this approach. Sorry for confusion

Comment (d3xt3r2909) [reply lib/src/server/linked_in_auth_code_widget.dart:27]: Marking this conversation thread as resolved.

**Comment by d3xt3r2909** [comment]
Conflicting because of https://github.com/d3xt3r2909/linkedin_login/pull/72/files
